### PR TITLE
Fix crash in generate constructor for C#

### PR DESCRIPTION
**Pull request by jmarolf**
Fixes https://github.com/dotnet/roslyn/issues/6920

### CODE REVIEW

**Comment by jmarolf** [comment]
@dotnet/roslyn-ide

File: src/Features/Core/Portable/GenerateMember/GenerateConstructor/GenerateConstructorHelpers.cs [hunk 1]
```diff
@@ -70,9 +70,9 @@ private static bool ParameterTypesMatch(SemanticDocument document, IList<ITypeSy
                 var type1 = parameterTypes[i];
                 if (type1 != null)
                 {
-                    var type2 = method.Parameters[i].Type;
+                    var type2 = method.Parameters[i]?.Type;
```
Comment (Pilchie) [inline src/Features/Core/Portable/GenerateMember/GenerateConstructor/GenerateConstructorHelpers.cs]: When do you get an `IMethodSymbol` with null's in it's `Parameters`?  That sounds like a bug in the compiler.

**Comment by dpoeschl** [comment]
:+1:

**Comment by davkean** [comment]
@Pilchie Can you look over the latest change?

**Comment by jasonmalinowski** [comment]
:+1:

**Comment by Pilchie** [comment]
:+1:

**Pull request merged**